src/gui/qgsattributeform.cpp

```diff
@@ -641,19 +638,20 @@ QString QgsAttributeForm::createFilterExpression() const
 void QgsAttributeForm::onAttributeChanged( const QVariant &value )
 {
   QgsEditorWidgetWrapper *eww = qobject_cast<QgsEditorWidgetWrapper *>( sender() );
-
   Q_ASSERT( eww );
 
+  const QVariant oldValue = mFeature.attribute( eww->fieldIdx() );
+
+  // Safetey check, if we receive the same value again, no reason to do anything
+  if ( oldValue == value && oldValue.isNull() == value.isNull() )
+    return;
+
   switch ( mMode )
   {
     case SingleEditMode:
     case AddFeatureMode:
     {
-      // don't emit signal if it was triggered by a feature change
-      if ( !mIsSettingFeature )
-      {
-        emit attributeChanged( eww->field().name(), value );
-      }
+      emit attributeChanged( eww->field().name(), value );
```

Comment:
@nyalldawson Do you know if it's actually required to send out this signal before updateJoinedFields here, the same signal is sent again further down the same function unconditionally. I've seen your name in git blame, so I thought I'd ask.

Reply:
I'll look - I can't recall off hand.